proposed/2023/project-exclusion-for-transitive-dependencies.md

```diff
@@ -0,0 +1,130 @@
+- GitHub PR https://github.com/NuGet/NuGet.Client/pull/5302
+
+## Summary
+
+Currently, when listing transitive packages using `dotnet list package --include-transitive`, both project and packages dependencies are reported. The goal of this proposal is to provide a way to filter the projects out of the report.
+
+## Motivation 
+
+Packages and projects are two vastly different types of references.
+Packages are published to a repository, allowing consumption by multiple pieces of software and repositories. Metadata are attached to packages or package versions, either included in the package, or added as a decoration - for deprecation or vulnerabilities for example -, which can be consumed to take decisions. On the contrary, projects are _local_ entities, living in a single repository, and do not have an existence outside of it.
+
+The `list package` command already supports filtering the results to retrieve only deprecated (`--deprecated`), vulnerable (`--vulnerable`), or not up-to-date (`--outdated`) packages. These commands only return actual packages, which is understandable as the underlying information used to choose what to report is managed by the _package_ manager, aka nuget.
+
+Allowing to remove projects from the output when none of these flags is provided would allow users to get an accurate view of their package dependencies. This would ensure that the output of the command can be processed in an uniform way by either a user, or a piece of software - through the machine readable output - to support any report, or process associated with packages, even those with no issues related to deprecation, vulnerabilities, or up-to-dateness.
+
+## Explanation
+
+### Functional explanation
+
+A new `--exclude-project` option will be added to the `list` command. This option will ensure that no project will be reported in the output when used with `--include-transitive`.
+
+Given a solution setup like the one below,
+```
+solution.sln
+├── projectA.csproj
+│   └── package1.nupkg (2.5.6)
+│       └── package2.nupkg (1.0.0)
+└── projectB.csproj
+    ├── projectA.csproj
+    └── package3.nupkg (30.0.1)
+        └── package4.nupkg (1.0.5)
+```
+
+Running `dotnet list solution.sln package --include-transitive` produces the following output:
+```
+Project 'projectA' has the following package references
+   [net7.0]:
+   Top-level Package      Requested   Resolved
+   > package1             2.5.6       2.5.6
+
+   Transitive Package       Resolved
+   > package2               1.0.0
+
+Project 'projectB' has the following package references
+   [net7.0]:
+   Top-level Package      Requested   Resolved
+   > package3             30.0.1      30.0.1
+
+   Transitive Package       Resolved
+   > package4               1.0.5
+   > projectA               1.0.0
+```
+
+Adding `--exclude-project` to the command would make `dotnet list solution.sln package --include-transitive --exclude-project` produce the following output:
```

Comment:
I think these 2 examples actually do summarize that we're having an inconsistent behavior and understanding that may be priority #1. 

If the command truly did list projects, `projectA` wouldn't be listed as transitive to projectB, it's a direct reference. I'm guessing it just couldn't have a root (?) and it got left there.